component/src/main/java/org/wso2/carbon/identity/authenticator/pinterest/PinterestAuthenticator.java

```diff
@@ -0,0 +1,394 @@
+						String subjectFromClaims = FrameworkUtils
+								.getFederatedSubjectFromClaims(context.getExternalIdP().getIdentityProvider(), claims);
+						associateSubjectFromClaims(context, subjectFromClaims, claims);
+					} else {
+						throw new AuthenticationFailedException("Claims for the user not found for access Token : " +
+						                                        accessToken);
+					}
+				} else {
+					throw new AuthenticationFailedException("Authentication Failed, access token is not available");
+				}
+			} catch (OAuthSystemException e) {
+				throw new AuthenticationFailedException(
+						"Exception while building a request to get the access token", e);
+			} catch (ApplicationAuthenticatorException e) {
+				throw new AuthenticationFailedException("Error while building the claim mapping", e);
+			}
+		} catch (OAuthProblemException e) {
+			throw new AuthenticationFailedException("Authentication Failed in OAuthClientResponse", e);
+		}
+	}
+
+	/**
+	 * This method is to get the Pinterest user details.
+	 *
+	 * @param url         user info endpoint.
+	 * @param accessToken access token.
+	 * @return user info
+	 * @throws ApplicationAuthenticatorException
+	 */
+	private JSONObject fetchUserInfo(String url, String accessToken) throws ApplicationAuthenticatorException {
+		if (log.isDebugEnabled()) {
+			log.debug("Sending the request for getting the user info");
+		}
+		StringBuilder stringBuilder = new StringBuilder();
+		BufferedReader bufferedReader = null;
+		HttpURLConnection httpConnection = null;
+		JSONObject jsonObj = null;
+		try {
+			URL obj = new URL(url + "?" + PinterestAuthenticatorConstants.ACCESS_TOKEN + "=" + accessToken);
+			URLConnection connection = obj.openConnection();
+			// Cast to a HttpURLConnection
+			if (connection instanceof HttpURLConnection) {
+				httpConnection = (HttpURLConnection) connection;
+				httpConnection.setConnectTimeout(PinterestAuthenticatorConstants.CONNECTION_TIMEOUT_VALUE);
+				httpConnection.setReadTimeout(PinterestAuthenticatorConstants.READ_TIMEOUT_VALUE);
+				httpConnection.setRequestMethod(PinterestAuthenticatorConstants.HTTP_GET_METHOD);
+				bufferedReader = new BufferedReader(new InputStreamReader(httpConnection.getInputStream()));
+			} else {
+				throw new ApplicationAuthenticatorException("Exception while casting the HttpURLConnection");
+			}
+			String inputLine = bufferedReader.readLine();
+			while (inputLine != null) {
+				stringBuilder.append(inputLine).append("\n");
+				inputLine = bufferedReader.readLine();
+			}
+			jsonObj = new JSONObject(stringBuilder.toString());
+		} catch (MalformedURLException e) {
+			throw new ApplicationAuthenticatorException("MalformedURLException while generating the user info URL: "
+			                                            + url, e);
+		} catch (ProtocolException e) {
+			throw new ApplicationAuthenticatorException("ProtocolException while setting the request method: " +
```

Comment:
Need to put the URL on the error message too.  There is no real value of an error log such as 
"ProtocolException while setting the request method: POST" without the offending URL.